chaosLib/litmus/poddelete/lib/podDelete.py

```
@@ -0,0 +1,167 @@
+def PreparePodDelete(experimentsDetails , resultDetails, eventsDetails, chaosDetails, clients):
+
+	#Waiting for the ramp time before chaos injection
+	if experimentsDetails.RampTime != 0 :
+		logging.info("[Ramp]: Waiting for the %s ramp time before injecting chaos",experimentsDetails.RampTime)
+		common.WaitForDuration(experimentsDetails.RampTime)
+	
+	# mode for chaos injection
+	if experimentsDetails.Sequence.lower() == "serial":
+		err = injectChaosInSerialMode(experimentsDetails, chaosDetails, eventsDetails, resultDetails, clients)
+		if err != None:
+			return err
+	elif experimentsDetails.Sequence.lower() == "parallel":
+		err = injectChaosInParallelMode(experimentsDetails, chaosDetails, eventsDetails, resultDetails, clients)
+		if err != None:
+			return err
+	else:
+		return ValueError("[Error]: {} sequence is not supported".format(experimentsDetails.Sequence))
+
+	#Waiting for the ramp time after chaos injection
+	if experimentsDetails.RampTime != 0 :
+		logging.info("[Ramp]: Waiting for the %s ramp time after injecting chaos",experimentsDetails.RampTime)
+		common.WaitForDuration(experimentsDetails.RampTime)
+
+	return None
+
+# injectChaosInSerialMode delete the target application pods serial mode(one by one)
+def injectChaosInSerialMode(experimentsDetails , chaosDetails , eventsDetails , resultDetails, clients): 
+	
+	#Initialising GracePeriod
+	GracePeriod = 0
+	
+	#ChaosStartTimeStamp contains the start timestamp, when the chaos injection begin
+	ChaosStartTimeStamp = datetime.now()
+	duration = (datetime.now() - ChaosStartTimeStamp).seconds
+	
+	while duration < experimentsDetails.ChaosDuration:
+		# Get the target pod details for the chaos execution
+		# if the target pod is not defined it will derive the random target pod list using pod affected percentage
+		if experimentsDetails.TargetPods == "" and chaosDetails.AppDetail.Label == "" :
+			return logging.warning("[Warning]: Please provide one of the appLabel or TARGET_PODS")
```

Can we return the error instead, we need to fail the experiment